src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -85,6 +97,22 @@ public void AllowStop()
 
         public void Stop(TimeSpan timeout)
         {
+            // Close and wait for all connections
+            ConnectionManager.WalkConnectionsAndClose();
+
+            // REVIEW: Should we use the timeout?
```

Comment:
Not timing out seems risky. Better for `Stop` to throw than hang here IMHO. If we change this, `WalkConnectionsAndClose`  would need to change for the same reason.